tests/Unit/Domain/Test_DomainCreators.cpp

```diff
@@ -224,3 +226,25 @@ SPECTRE_TEST_CASE("Unit.Domain.DomainCreators.TestHelperFunctions",
   tnsr::I<double, 3, Frame::Logical> expected{{{-1.0, -1.0, 1.0}}};
   CHECK(point_in_neighbor_frame<3>(custom_orientation, test_point) == expected);
 }
+
+SPECTRE_TEST_CASE("Unit.Domain.DomainCreators.Sphere.Separation",
```

Comment:
This should test that the domain is constructed as expected: check neighbors, external boundaries and the maps.